### PR TITLE
Sort clients in `minetest.get_server_status` and privs in `minetest.privs_to_string`

### PR DESCRIPTION
**Pull request by zmv7**
This PR applies alphabetical sorting to the `minetest.get_server_status` and `minetest.privs_to_string` functions and also to the output of `/haspriv` command.

## To do

This PR is Ready for Review.

## How to test

* Privs: test with using `/privs`, `/grant[me]`, `/revoke[me]` commands
* Clients: make some test server and connect some real or dummy clients with different names and check the ordering in the output of `/status` and `/haspriv` commands.

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -3182,17 +3182,28 @@ std::string Server::getStatusString()
 	os << " | clients: ";
 	if (m_env) {
 		std::vector<session_t> clients = m_clients.getClientIDs();
+
```
Comment (appgurueu) [inline src/server.cpp:3185]: Why not use `m_clients.getPlayerNames()` directly?

File: src/server.cpp [hunk 2]
```diff
@@ -3182,17 +3182,28 @@ std::string Server::getStatusString()
...
+		// Create vector to store player names
+		std::vector<std::string> playerNames;
```
Comment (appgurueu) [inline src/server.cpp:3187]: `player_names` btw

Comment (SmallJoker) [reply src/server.cpp:3187]: "Why?", you might ask. https://dev.minetest.net/Code_style_guidelines#Miscellaneous -> "snake_case for local variables".

Comment (zmv7) [reply src/server.cpp:3187]: Ahh, got it

File: src/server.cpp [hunk 3]
```diff
@@ -3181,14 +3181,14 @@ std::string Server::getStatusString()
 	bool first = true;
 	os << " | clients: ";
 	if (m_env) {
-		std::vector<session_t> clients = m_clients.getClientIDs();
-		for (session_t client_id : clients) {
-			RemotePlayer *player = m_env->getPlayer(client_id);
+		// Get the list of player names
```
Comment (appgurueu) [inline src/server.cpp:3184]: comments like this can go, the following line of code is self explaining

**Review by appgurueu — APPROVED** [review]
Looks fine

File: src/server.cpp (hunk 3 above)
Comment (sfan5) [reply src/server.cpp:3184]:
```suggestion
```

File: src/server.cpp [hunk 4]
```diff
@@ -3181,14 +3181,13 @@ std::string Server::getStatusString()
...
 	os << " | clients: ";
 	if (m_env) {
-		std::vector<session_t> clients = m_clients.getClientIDs();
-		for (session_t client_id : clients) {
-			RemotePlayer *player = m_env->getPlayer(client_id);
+		std::vector<std::string> player_names = m_clients.getPlayerNames();
 
-			// Get name of player
-			const std::string name = player ? player->getName() : "<unknown>";
+		// Sort player names alphabetically
```
Comment (sfan5) [inline src/server.cpp:3186]:
```suggestion
		// Sort player names alphabetically
```

File: src/server.cpp [hunk 5]
```diff
@@ -3181,14 +3181,13 @@ std::string Server::getStatusString()
...
+		std::sort(player_names.begin(), player_names.end());
 
-			// Add name to information string
+		// Concatenate sorted names to the output string
```
Comment (sfan5) [inline src/server.cpp:3189]:
```suggestion
		// Concatenate sorted names to the output string
```

**Review by sfan5 — APPROVED** [review]

**Pull request merged**